### ISSUE TITLE
Cant load or save collection of objects

### ISSUE THREAD

**Issue by jportway** [pureconfig/pureconfig#7]
Hi,
PureConfig is brilliant, but I'm having trouble loading or saving Lists or Maps of objects.
I don't think I'm doing anything wrong.

Here's a failing test case:

``` scala
  it should "save a list of objects properly" in {
    case class DeepConfig(list:List[Wibble])
    case class Wibble(x:String,y:String)
    withTempFile { (configFile: Path) =>
      val w1=Wibble("X","y")
      val w2=Wibble("hello","goodbye")
      val expectedConfig = DeepConfig(List(w1,w2))
      saveConfigAsPropertyFile(expectedConfig, configFile, overrideOutputPath = true)
      // println(scala.io.Source.fromFile(configFile.toFile).mkString)
      val config = loadConfig[DeepConfig](configFile)

      config should be(Success(expectedConfig))
    }
  }
```

the output file looks like this :

```
list.tl$1.tl$1=""
list.tl$1.head.x="hello"
list.head.y="y"
list.tl$1.head.y="goodbye"
list.head.x="X"
```

**Comment by melrief** [comment]
This is a bug indeed but a different one: the idea for collections, for now, is that they can be loaded/saved iff the `T` on which they are parametrized has an instance of `StringConvert`. Classes like `Wibble` have no instance like that and this means that something strange is happening with `List`. If, for instance, you change `List` to `Stream` in `DeepConfig` you will get a compilation error.

I understand the idea of saving/loading list of objects but for now it's not yet supported because it would require more layers. The current system just allow for saving collections of "primitives" types, aka types with a `StringConvert` instance like `Int` and `String`.

About the error itself, or why scalac can save/load `DeepConfig` instances, Idea is telling me that the scalac implicit resolution for it  is `deriveInstanceWithGeneric` and that means that there is an actual `Generic` representation of `List[T]` where `T` is an object. I'm not sure why but looks like `List` is a special case of collection.

I will investigate more, in the meanwhile thanks for pointing out this!

**Comment by melrief** [comment]
I confirm that the problem is that `List` has a `Generic` instance for Shapeless and this breaks the current code:

``` scala
scala> val g = Generic[List[Int]]
g: shapeless.Generic[List[Int]]{type Repr = shapeless.:+:[scala.collection.immutable.::[Int],shapeless.:+:[scala.collection.immutable.Nil.type,shapeless.CNil]]} = fresh$macro$2$1@40af21d9
```

The solution that I see is to push a `ConfigConvert` that takes in account also collections of objects and allow to save/load collection of objects.

**Comment by melrief** [comment]
@jportway I've released pureconfig 0.1.4 with support for collections of objects. There are [some tests](https://github.com/melrief/pureconfig/blob/master/src/test/scala/pureconfig/PureconfSuite.scala#L168) showing it in action. If you have sometime to dedicate to pureconfig, consider testing again the library with your example and see if it works. For now, thank you for finding the bug!
